src/language/concurrency.md

```diff
@@ -16,87 +19,111 @@ prevpage:
+The event loop is responsible for executing your program's code,
+collecting and processing events, and more.
+
+As your application runs, all events are added to a queue,
+called the *event queue*.
+Events can be anything from requests to repaint the UI,
+to user taps and keystrokes, to I/O from the disk.
+Because your app can’t predict in what order events will happen,
+the event loop processes events in the order they're queued, one at a time.
+
+![A figure showing events being fed, one by one, into the
+event loop](/assets/img/language/concurrency/event-loop.png)
+
+The way the event loop functions resembles this code:
+
+```dart
+while (eventQueue.waitForEvent()) {
+  eventQueue.processNextEvent();
+}
+```
+
+This example event loop is synchronous and runs on a single thread.
+However, most Dart applications need to do more than one thing at a time. 
+For example, a client application might need to execute an HTTP request, 
+while also listening for a user to tap a button. 
+To handle this, Dart offers many async APIs, 
+like [Futures, Streams, and async-await](/language/async).
+These APIs are built around this event loop.
+
+For example, consider making a network request:
+
+```dart
+http.get('https://example.com').then((response) {
+  if (response.statusCode == 200) {
+    print('Success!')'
+  }  
+}
+```
 
-[isolates section]: #how-isolates-work
+When this code reaches the event loop, it immediately calls the
+first clause, `http.get`, and returns a `Future`.
+It also tells the event loop to hold onto the callback in the `then()` clause
+until the HTTP request resolves. When that happens, it should
+execute that callback, passing the result of the request as an argument.
 
+![Figure showing async events being added to an event loop and
+holding onto a callback to execute later
+.](/assets/img/language/concurrency/async-event-loop.png)
 
-### Future and Stream types
+This same model is generally how the event loop handles all other
+asynchronous events in Dart, such as [`Stream`][] objects.
 
-The Dart language and libraries use `Future` and `Stream` objects to
-represent values to be provided in the future.
-For example, a promise to eventually provide an `int` value
-is typed as `Future<int>`.
-A promise to provide a series of `int` values
-has the type `Stream<int>`.
+[`Stream`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/Stream-class.html
 
-As another example, consider the dart:io methods for reading files.
-The synchronous `File` method [`readAsStringSync()`][]
-reads a file synchronously,
-blocking until the file is either fully read or an error occurs.
-The method then either returns an object of type `String`
-or throws an exception.
-The asynchronous equivalent, [`readAsString()`][],
-immediately returns an object of type `Future<String>`.
-At some point in the future,
-the `Future<String>` completes with either a string value or an error.
+## Types and syntax for asynchronous programming
```

Comment:
Once I got to the `## Isolates` section, I started thinking about how these sections will probably be used as the link to "define" async and isolates around the site, so maybe a more definitive section title would be better here. Also, one that mirrors the `## Isolates` section since they're kind of sister sections.

```suggestion
## Asynchronous programming
```

And then add the "types and syntax" thing to the intro sentence (commented below)